tools/snap/build_remote.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python3
+import argparse
+import glob
+import datetime
+from multiprocessing import Pool, Process, Manager
+import re
+import subprocess
+import sys
+import time
+from os.path import join, realpath, dirname, basename
+
+
+CERTBOT_DIR = dirname(dirname(dirname(realpath(__file__))))
+PLUGINS = [basename(path) for path in glob.glob(join(CERTBOT_DIR, 'certbot-dns-*'))]
+
+
+def _execute_build(target, archs, status, workspace):
+    process = subprocess.Popen([
+        'snapcraft', 'remote-build', '--launchpad-accept-public-upload', '--recover', '--build-on', ','.join(archs)
+    ], stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True, cwd=workspace)
+
+    line = process.stdout.readline()
+    while line:
+        _extract_state(target, line, status)
+        line = process.stdout.readline()
+
+
+def _build_snap(target, archs, status):
+    status[target] = {arch: '...' for arch in archs}
+
+    if target == 'certbot':
+        workspace = CERTBOT_DIR
+    else:
+        workspace = join(CERTBOT_DIR, target)
+        subprocess.check_output(
+            ('"{0}" tools/strip_hashes.py letsencrypt-auto-source/pieces/dependency-requirements.txt '
+             '| grep -v python-augeas > "{1}/snap-constraints.txt"').format(sys.executable, workspace),
+            shell=True, cwd=CERTBOT_DIR)
+
+    retry = 3
+    while retry:
+        _execute_build(target, archs, status, workspace)
+        if not [status[target] for arch in archs if status[target][arch] == '...']:
```

Comment:
This is just checking that the launchpad build was at least queued for each architecture right? If so, can you add a comment explaining this? I wasn't initially sure what this code did.

One thing we may be able to do instead though is rely on the exit status of `snapcraft remote-build`. It seems to return 0 even if the build fails and I'd expect it to return something else if it got interrupted.